### PR TITLE
Make UserAccountsDrawerHeader accessible

### LINKED ISSUE #12379: Make UserAccountsDrawerHeader accessible
Compare it with the OEM widget and provide similar a11y support.

### LINKED ISSUE #13743: Regression in UserAccountsDrawerHeader layout
https://github.com/flutter/flutter/pull/13711 caused a regression in the layout of user user name and email in the user account switcher:

user-accounts-drawer-header

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes #13743. 
Fixes #12379.
Follow-up to #13745

Also adds an option to hide gestures introduced by `InkWell` and `InkResponse` from the semantics tree (see also `GestureDetector.excludeFromSemantics`).

/cc @yjbanov @tvolkert

### CODE REVIEW

**Comment by tvolkert** [comment]
RSLGTM

File: packages/flutter/lib/src/material/user_accounts_drawer_header.dart [hunk 1]
```diff
@@ -67,66 +74,130 @@ class _AccountDetails extends StatelessWidget {
...
-        padding: const EdgeInsets.symmetric(vertical: 8.0),
-        child: new Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-          children: (accountEmailLine != null && accountNameLine != null)
-            ? <Widget>[accountNameLine, accountEmailLine]
-            : <Widget>[accountNameLine ?? accountEmailLine]
+    if (onTap != null) {
+      final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+      final Widget dorpDownIcon = new LayoutId(
```
Comment (Hixie) [inline packages/flutter/lib/src/material/user_accounts_drawer_header.dart]: drop

**Review by yjbanov — APPROVED** [review]

**Comment by Hixie** [comment]
InkWell change looks good though it could do with a dedicated test (or at least, comments in both the ink_well_test.dart file and the user_accounts_drawer_header_test.dart file pointing to each other, to make it clear that it is tested somewhere).

LGTM

File: packages/flutter/lib/src/material/user_accounts_drawer_header.dart [hunk 2]
```diff
@@ -67,66 +74,130 @@ class _AccountDetails extends StatelessWidget {
...
+  static final String accountName = 'accountName';
+  static final String accountEmail = 'accountEmail';
+  static final String dropdownIcon = 'dropdownIcon';
+
+  @override
+  void performLayout(Size size) {
+    Size iconSize;
+
+    if (hasChild(dropdownIcon)) {
+      // place the dropdown icon in bottom right
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/lib/src/material/user_accounts_drawer_header.dart]: Good catch! We indeed need to manually handle RTL here. Updated the PR with code for that.

**Comment by goderbauer** [comment]
Added dedicated tests for InkWell and updated PR to support RTL properly. Please take another look.

**Comment by yjbanov** [comment]
lgtm

**Pull request merged**